src/abstractdataframe/reshape.jl

```diff
@@ -164,8 +164,8 @@ function stack(df::AbstractDataFrame,
         # (note that copyto! inserts levels in their order of appearance)
         nms = names(df, ints_measure_vars)
         simnms = similar(nms, variable_eltype)
-        catnms = simnms isa Vector ? PooledArray(catnms) : simnms
-        copyto!(catnms, nms)
+        copyto!(simnms, nms)
+        catnms = simnms isa Vector ? PooledArray(simnms) : simnms
```

Comment:
Why call `copyto!(simnms, nms)` rather than `copyto!(catnms, nms)` later? Wouldn't the latter be faster?

Reply:
Without https://github.com/JuliaData/PooledArrays.jl/pull/82 released this will not work. After I make a release of PooledArrays.jl I will make this change.

Reply:
I have changed this as indeed it is slightly faster. Here using `copyto!` should be safe as we know `nms` cannot have `#undef` so we are not affected by https://github.com/JuliaLang/julia/issues/45125.